editor/src/main/java/org/pentaho/pms/ui/util/Splash.java

```diff
@@ -42,48 +41,133 @@
 
 /**
  * Displays the splash and about screen
- *
  */
 public class Splash {
 
   private static final Log logger = LogFactory.getLog( Splash.class );
 
   private static final String FONT_TYPE = "Helvetica";
-
   private static final String LICENSE_FILE_PATH = "./LICENSE.TXT";
 
+  private boolean isStartup;
+  private boolean isMouseDown = false;
+  private boolean isMouseMoveInProgress = false;
+  private int xPos = 0;
+  private int yPos = 0;
+
   private Shell shell;
   private Font verFont;
   private Font licFont;
+  private Display display;
+  private Image aboutScreenImage;
 
   private int textHorizontalPosition = 300;
 
 
   public Splash( Display display ) {
-    this( display, new Shell( display, SWT.APPLICATION_MODAL ) );
+    this( display, new Shell( display, SWT.APPLICATION_MODAL ), true );
   }
 
   protected Splash( Display display, Shell splashShell ) {
+    this( display, splashShell, false );
+  }
 
-    Rectangle displayBounds = display.getPrimaryMonitor().getBounds();
+  protected Splash( Display display, Shell splashShell, boolean isStartup ) {
+    this.display = display;
+    this.isStartup = isStartup;
 
     verFont = new Font( display, FONT_TYPE, 14, SWT.BOLD );
 
     shell = splashShell;
 
-    Image aboutScreenImage = GUIResource.getInstance().getImageMetaSplash();
+    aboutScreenImage = GUIResource.getInstance().getImageMetaSplash();
     shell.setImage( aboutScreenImage );
-
     shell.setText( Messages.getString( "MetaEditor.USER_HELP_METADATA_EDITOR" ) );
 
+    setupShellDimensions();
+
+    setupShellListeners();
+
+    shell.open();
+
+    if ( isStartup ) { //show splash for 5s on startup
+      long endTime = System.currentTimeMillis() + 5000;
+      while ( !shell.isDisposed() && endTime > System.currentTimeMillis() ) {
+        if ( isMacOS() && !display.readAndDispatch() ) {
+          display.sleep();
+        }
+      }
+      shell.dispose();
+    }
+  }
+
+  private void setupShellDimensions() {
+    Rectangle displayBounds = display.getPrimaryMonitor().getBounds();
+    Rectangle bounds = aboutScreenImage.getBounds();
+    int x = ( displayBounds.width - bounds.width ) / 2;
+    int y = ( displayBounds.height - bounds.height ) / 2;
+
+    shell.setSize( bounds.width, bounds.height );
+    shell.setLocation( x, y );
+  }
+
+  private void setupShellListeners() {
+    setupShellPaintListener();
+
+    shell.addMouseListener( new MouseListener() {
+
+      @Override
+      public void mouseUp( MouseEvent e ) {
```

Comment:
Not a blocker but for future developers, I would document the motivation for this logic for need for `addMouseListener()` and the interaction between `isMouseDown` and  `isMouseMoveInProgress`

Reply:
Same deal as setupShellDimensions method, this code was in AboutDialog, I just moved it back here as I wanted to have these listeners set on any splash instance so you could close it faster than 5s on startup if you wanted to. Moved all listener setup out of constructor to reduce CC (Sonar smell)